src/general/ModLoader.cs

```diff
@@ -0,0 +1,320 @@
+using Godot;
+using Newtonsoft.Json;
+using Directory = System.IO.Directory;
+using File = System.IO.File;
+
+public class ModLoader : Control
+{
+    [Export]
+    public NodePath UnloadedItemListPath;
+
+    [Export]
+    public NodePath ModInfoNamePath;
+
+    [Export]
+    public NodePath ModInfoAuthorPath;
+
+    [Export]
+    public NodePath ModInfoVersionPath;
+
+    [Export]
+    public NodePath ModInfoDescriptionPath;
+
+    [Export]
+    public NodePath LoadedItemListPath;
+
+    [Export]
+    public NodePath ConfirmationPopupPath;
+
+    private ItemList unloadedItemList;
+    private ItemList loadedItemList;
+
+    // Labels For The Mod Info Box
+    private Label modInfoName;
+    private Label modInfoAuthor;
+    private Label modInfoVersion;
+    private Label modInfoDescription;
+
+    private ConfirmationDialog confirmationPopup;
+
+    private List<ModInfo> modList = new List<ModInfo>();
+    private List<ModInfo> loadedModList = new List<ModInfo>();
+
+    [Signal]
+    public delegate void OnModLoaderClosed();
+
+    public override void _Ready()
+    {
+        unloadedItemList = GetNode<ItemList>(UnloadedItemListPath);
+        loadedItemList = GetNode<ItemList>(LoadedItemListPath);
+
+        modInfoName = GetNode<Label>(ModInfoNamePath);
+        modInfoAuthor = GetNode<Label>(ModInfoAuthorPath);
+        modInfoVersion = GetNode<Label>(ModInfoVersionPath);
+        modInfoDescription = GetNode<Label>(ModInfoDescriptionPath);
+
+        confirmationPopup = GetNode<ConfirmationDialog>(ConfirmationPopupPath);
+
+        DirectoryInfo modFolder = Directory.CreateDirectory($"{Directory.GetCurrentDirectory()}\\mods");
+        foreach (DirectoryInfo currentMod in modFolder.EnumerateDirectories())
+        {
+            if (!File.Exists(currentMod.FullName + "/mod_info.json"))
+            {
+                continue;
+            }
+
+            var currentModInfo =
+                JsonConvert.DeserializeObject<ModInfo>(ReadJSONFile(currentMod.FullName + "/mod_info.json"));
+
+            if (currentModInfo.AutoLoad)
+            {
+                continue;
+            }
+
+            currentModInfo.Location = currentMod.FullName;
+            modList.Add(currentModInfo);
+            unloadedItemList.AddItem(currentModInfo.ModName);
+        }
+    }
+
+    // Copied From The PauseMenu.cs
+    private static string ReadJSONFile(string path)
+    {
+        using (var file = new Godot.File())
+        {
+            file.Open(path, Godot.File.ModeFlags.Read);
+            var result = file.GetAsText();
+
+            // This might be completely unnecessary
+            file.Close();
+
+            return result;
+        }
+    }
+
+    private void OnUnloadedModSelected(int index)
+    {
+        var tempModInfo = modList[index];
+        modInfoName.Text = tempModInfo.ModName;
+        modInfoAuthor.Text = "Author: " + tempModInfo.Author;
+        modInfoVersion.Text = "Version: " + tempModInfo.Version;
```

Comment:
Splitting these into separate controls would be preferrable to allow for translations. The author value loaded from the mod should be in a separate label by itself.